plugins/repository-s3/src/test/java/org/elasticsearch/repositories/s3/AmazonS3Fixture.java

```diff
@@ -312,6 +330,40 @@ public static void main(final String[] args) throws Exception {
             return newInternalError(request.getId(), "Something is wrong with this POST multiple deletes request");
         });
 
+        // non-authorized requests
+
+        Function<String, Response> credentialResponseFunction = prefix -> {
+            final Date expiration = new Date(new Date().getTime() + TimeUnit.DAYS.toMillis(1));
+            final String response = "{"
+                 + "\"AccessKeyId\": \"" + prefix + "_KEYID" + "\","
+                 + "\"Expiration\": \"" + DateUtils.formatISO8601Date(expiration) + "\","
+                 + "\"RoleArn\": \"" + prefix + "_ROLE" + "\","
+                 + "\"SecretAccessKey\": \"" + prefix + "_SCR_KEY" + "\","
+                 + "\"Token\": \"" + prefix + "_TKN" + "\""
+                 + "}";
+
+            final Map<String, String> headers = new HashMap<>(contentType("application/json"));
+            return new Response(RestStatus.OK.getStatus(), headers, response.getBytes(UTF_8));
+        };
+
+        // GET
+        //
+        // http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html
+        handlers.insert("* GET /latest/meta-data/iam/security-credentials/", (request) -> {
+            final String response = "securitycredentials42";
+
+            final Map<String, String> headers = new HashMap<>(contentType("text/plain"));
+            return new Response(RestStatus.OK.getStatus(), headers, response.getBytes(UTF_8));
+        });
+
+        // GET
+        //
+        // http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html
+        handlers.insert("* GET /latest/meta-data/iam/security-credentials/{credentials}", (request) -> {
+            final String credentials = request.getParam("credentials");
+            return credentialResponseFunction.apply(credentials);
```

Comment:
On deeper thought, this seems unduly lenient: it should only return credentials for the role that `GET /latest/meta-data/iam/security-credentials/` returned, and should return 404 otherwise.

Also I think `credentialResponseFunction` can be inlined, it's only used in one place.

Also also we could prevent cheating slightly more by inventing random credentials when the service starts up, rather than synthesising them from the role name.

Reply:
I extract it for doing ECS part as well - it has to return similar json,

for randomization ... I bit confused here in terms of how to reproduce it in case of failure

Reply:
@DaveCTurner s3 perm/temp authorization tokens have to be randomized as well - should we ?

Reply:
I think we cannot (yet) do this with auth tokens received externally because they need to be passed to the test suite as well as this fixture, so we've had to settle on just using the same (long) string in both places. It would indeed be nicer if there were no magic strings in the fixture at all.

For deterministic testing, maybe you can pass a seed in from Gradle somehow? Or just hard-code them, it's not that important. I'd rather they weren't synthesised as they are now.

Reply:
it sounds like it is better to provide some kind of properties file that contains tokens, paths etc - esp. keeping in mind coming ECS test

Reply:
Sure, good plan.